code/src/cljs/sixsq/nuvla/ui/edges/views.cljs

```diff
@@ -178,34 +181,26 @@
                                        :state-selector-subs      :sixsq.nuvla.ui.edges.subs/state-selector
                                        :stacked?                 true}]]]]))))
 
-(defn switch-from-cluster-view?
-  [current-view new-view]
-  (when (= current-view :cluster)
-    (dispatch [::pagination-plugin/change-page
-               [::spec/pagination] 1]))
-  (reset! view-type new-view))
+(def view->icon-classes
+  {spec/cards-view   "grid layout"
+   spec/table-view   "table"
+   spec/map-view     "map"
+   spec/cluster-view "fas fa-chart-network"})
 
 (defn MenuBar []
-  (let [loading? (subscribe [::subs/loading?])]
+  (let [loading?  (subscribe [::subs/loading?])
+        view-type (subscribe [::subs/view-type])]
     (fn []
       [components/StickyBar
        [ui/Menu {:borderless true, :stackable true}
         [views-utils/AddButton]
-        [ui/MenuItem {:icon     "grid layout"
-                      :active   (= @view-type :cards)
-                      :on-click #(switch-from-cluster-view? @view-type :cards)}]
-        [ui/MenuItem {:icon     "table"
-                      :active   (= @view-type :table)
-                      :on-click #(switch-from-cluster-view? @view-type :table)}]
-        [ui/MenuItem {:icon     "map"
-                      :active   (= @view-type :map)
-                      :on-click #(switch-from-cluster-view? @view-type :map)}]
-        [ui/MenuItem {:active   (= @view-type :cluster)
-                      :on-click #(do
-                                   (dispatch [::pagination-plugin/change-page
-                                              [::spec/pagination] 1])
-                                   (reset! view-type :cluster))}
-         [ui/Icon {:className "fas fa-chart-network"}]]
+        (doall
+          (for [view spec/view-types]
+
+            ^{:key view}
+            [ui/MenuItem {:active   (= @view-type view)
+                          :on-click #(switch-view! view)}
+             [ui/Icon {:className (view->icon-classes view)}]]))
```

Comment:
Isn't you that pushed to replace with :class :P

Reply:
I just use `:class` when I write it myself, but usually don't bother changing it. This one already existed so I didn't bother.

But now I'm curious how it came to be known that I pushed for this replacement 😄